docs/reference/mapping/dynamic/templates.asciidoc

```diff
@@ -4,7 +4,7 @@
 Dynamic templates allow you to define custom mappings that can be applied to
 dynamically added fields based on:
 
-* the <<dynamic-mapping,datatype>> detected by Elasticsearch, with <<match-mapping-type,`match_mapping_type`>>.
+* the <<mapping-types,datatype>> detected by Elasticsearch, with <<match-mapping-type,`match_mapping_type`>>.
```

Comment:
I'm not sure about this change, the point of this entry I think is to explain that the template can use the "match_mapping_type" rules to apply certain mapping types based on the datatype detected by the json parser. This might not necessarily be the type that dynamic mapping would detect, although I'm not too familiar with the process and how much they are similar. In each case, I think a link to the dynamic-mapping page might be here on purpose, maybe the whole first part of the sentence needs slight tweaking? I wonder what @jrodewig thinks about the way this is currently phrased

Reply:
I agree with @cbuescher. The field mapping types  are  **not** the same datatypes used by the `match_mapping_type` rules. I think it might be more helpful to remove that link entirely here:

```suggestion
* the <<match-mapping-type, match mapping datatype>> detected by {es}'s built-in JSON parser.
```